software/hwloc/v2.0/index.php

```diff
@@ -1,109 +1,16 @@
 <?php
 $topdir = "../../..";
 
-include_once("$topdir/software/hwloc/v2.0/version.inc");
-$title = "Portable Hardware Locality (hwloc): Version $ver_v2_0";
+include_once("version.inc");
+$title = "Portable Hardware Locality (hwloc): Version $release_series";
 
 include_once("$topdir/projects/hwloc/nav.inc");
 include_once("$topdir/includes/header.inc");
 include_once("$topdir/includes/downloads.inc");
 
-$md5 = read_checksums("downloads/md5sums.txt");
-$sha1 = read_checksums("downloads/sha1sums.txt");
-
-#############################################################################
-#
-# Current stable release
-#
-#############################################################################
-
-$t_stable = new downloadTable("./downloads", "./downloads",
-			      dirname($_SERVER["PHP_SELF"]), 0);
-
-if (isset($t_stable)) {
-    $src = "Version $ver_v2_0";
-    $names = array("hwloc-$ver_v2_0.tar.gz", "hwloc-$ver_v2_0.tar.bz2",
-		   "hwloc-win32-build-$ver_v2_0.zip", "hwloc-win64-build-$ver_v2_0.zip");
-    foreach ($names as $key => $n) {
-        if (file_exists("downloads/$n")) {
-            $t_stable->addFile($src, $n, $md5[$n], $sha1[$n]);
-        }
-    }
-}
-
-#############################################################################
-#
-# Prereleases
-#
-#############################################################################
-
-# See if there's a snapshot file that does not have any letters in its
-# version number (i.e., is not a stable release) and has a
-# corresponding .tar.bz2 file or .tar.gz here in the downloads
-# directory.
-
-$fp = fopen("downloads/latest_snapshot.txt", "r");
-$v = fgets($fp);
-fclose($fp);
-$v = substr($v, 0, strlen($v) - 1);
-
-if (preg_match("/[a-z]/i", $v)) {
-    $t_prerelease = new downloadTable("./downloads", "./downloads",
-                                      dirname($_SERVER["PHP_SELF"]), 0);
-
-    $s = "<div align=\"center\">$v<br />(pre-release)</div>";
-    $names = array("hwloc-$v.tar.gz", "hwloc-$v.tar.bz2",
-                   "hwloc-win32-build-$v.zip", "hwloc-win64-build-$v.zip");
-
-    $found = 0;
-    foreach ($names as $key => $n) {
-        if (file_exists("downloads/$n")) {
-            $t_prerelease->addFile($s, $n, $md5[$n], $sha1[$n]);
-            $found = 1;
-        }
-    }
-    if (!$found) {
-        unset($t_prerelease);
-    }
-}
-
-#############################################################################
-#
-# Older releases
-#
-#############################################################################
-
-$t_older = new downloadTable("./downloads", "./downloads",
-			     dirname($_SERVER["PHP_SELF"]), 0);
-
-if (isset($t_older)) {
-    $versions = array("2.0.0");
-
-    foreach ($versions as $key => $ver) {
-        $src = "Version $ver";
-        if ($ver == "2.0.0") {
-           $src .= "<br/><b><font color=\"red\">Broken ABI, do not use!</font></b>";
```

Comment:
Is there any way to add a custom warning (like the above for hwloc 2.0.0) with your generic code? Or should I make it a warning at the bottom of the page right below 2.0.0 in the table?

Reply:
I missed that; I could probably add a way to add that note, but if you're ok with a note at the bottom of the page, that would save much time for a feature that we've wanted once in 10 years :).

Reply:
Sure, that's OK